addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1536,9 +1539,28 @@ options.registry.WebsiteFormFieldModel = DisableOverlayButtonOption.extend({
 // Disable delete button for model required fields
 options.registry.WebsiteFormFieldRequired = DisableOverlayButtonOption.extend({
     start: function () {
-        this.disableButton('remove', _t('This field is mandatory for this Action. You cannot remove it.'));
+        this.disableButton("remove", _t(
+            "This field is mandatory for this action. You cannot remove it. Try hiding it with the"
+            + " 'Visibility' option instead and add it a default value."
+        ));
```

Comment:
I wonder if we should also prevent the user from changing the "input type." It doesn't make sense to be able to switch to "telephone," for example. While we're in the master branch, shouldn't we do something about it in this PR?

cc @pierre-pde

Reply:
@bvr-odoo , I think "input type" is not a great risk. 
The most important is that people can't change "type" for mandatory fields.
In the backend, nothing stops you to put something stupid in the email field :) 
![image](https://github.com/odoo/odoo/assets/75257428/fe206e16-7cd8-4ccb-97aa-756d997a2ae3)

Reply:
In the backend an input is an input, there is no notion of type, right?
In the website forms, the type will alter what you can type inside. The validation of the form depends of it.
For instance, if you type "aze" in the mail field, it will tell you it's wrong. But if you change it from "email" to "phone", it will let you type it.
I don't see a reason to let the user change that, we could even hide it IMO, which would make the UX/UI cleaner since it will be one option less.

All in all, if we keep the possibility to change that it's not the end of the world, but it does not makes sense AFAIC

Reply:
Ok I will add a commit on this PR to hide the input type selector for model required fields :+1: